drizzlepac/hlautils/svm_quality_analysis.py

```diff
@@ -38,22 +38,114 @@
 from astropy.stats import sigma_clipped_stats
 from astropy.table import Table
 import numpy as np
+from scipy.spatial import KDTree
 
 # Local application imports
-from drizzlepac.hlautils import astrometric_utils
+from drizzlepac.hlautils import astrometric_utils as au
 import drizzlepac.hlautils.diagnostic_utils as du
 import drizzlepac.devutils.comparison_tools.compare_sourcelists as csl
 from stsci.tools import logutil
-
+from stwcs.wcsutil import HSTWCS
 
 __taskname__ = 'svm_quality_analysis'
 
 MSG_DATEFMT = '%Y%j%H%M%S'
 SPLUNK_MSG_FORMAT = '%(asctime)s %(levelname)s src=%(name)s- %(message)s'
 log = logutil.create_logger(__name__, level=logutil.logging.NOTSET, stream=sys.stdout,
                             format=SPLUNK_MSG_FORMAT, datefmt=MSG_DATEFMT)
+# ----------------------------------------------------------------------------------------------------------------------
+
+def characterize_gaia_distribution(hap_obj, log_level=logutil.logging.NOTSET):
+    """Statistically describe distribution of GAIA sources in footprint.
+
+    Computes and writes the file to a json file:
+
+    - Number of GAIA sources
+    - X centroid location
+    - Y centroid location
+    - X offset of centroid from image center
+    - Y offset of centroid from image center
+    - X standard deviation
+    - Y standard deviation
+    - minimum closest neighbor distance
+    - maximum closest neighbor distance
+    - mean closest neighbor distance
+    - standard deviation of closest neighbor distances
+
+    Parameters
+    ----------
+    hap_obj : drizzlepac.hlautils.Product.FilterProduct
+        hap product object to process
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the .log file.
+        Default value is 'NOTSET'.
+
+    Returns
+    -------
+    Nothing
+    """
+    log.setLevel(log_level)
+
+    # get table of GAIA sources in footprint
+    gaia_table = generate_gaia_catalog(hap_obj, columns_to_remove=['mag', 'objID', 'GaiaID'])
+
+    # if log_level is either 'DEBUG' or 'NOTSET', write out GAIA sources to DS9 region file
+    if log_level <= 10:
+        reg_file = "{}_gaia_sources.reg".format(hap_obj.drizzle_filename[:-9])
```

Comment:
Instead of 10, please use the logging capabilities (logutil.logging.DEBUG).

Reply:
Issue resolved in commit [9441273](https://github.com/spacetelescope/drizzlepac/pull/636/commits/9441273a6ec8db6b9747c72bad4d78568b134e00)